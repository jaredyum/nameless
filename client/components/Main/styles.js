export default theme => ({
  main: {
    flex: 1,
    display: 'flex',
    flexDirection: 'column',
    maxWidth: '100%',
    [theme.breakpoints.up('md')]: {
      maxWidth: 'calc(100% - 240px)'
    }
  },
  toolbar: theme.mixins.toolbar
});
