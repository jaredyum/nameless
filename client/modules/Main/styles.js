export default theme => ({
  main: {
    flex: 1,
    display: 'flex',
    flexDirection: 'column'
  },
  toolbar: theme.mixins.toolbar
});
