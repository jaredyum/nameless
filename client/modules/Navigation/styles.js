/**
 * The default width of the drawer (sidebar).
 * @const {number}
 */
const DRAWER_WIDTH = 240;

/**
 * The parent styles for this modules.
 * @param {!Object} theme The MUI theme object.
 * @return {!Object.<string, !Object>} The generated styles.
 */
export default theme => ({
  active: {
    fontWeight: 'bold'
  },
  appBar: {
    zIndex: theme.zIndex.drawer + 1
  },
  drawerPaper: {
    width: DRAWER_WIDTH,
    [theme.breakpoints.up('md')]: {
      position: 'relative'
    }
  },
  flex: {
    flex: 1
  },
  link: {
    color: '#333',
    cursor: 'pointer',
    width: '100%',
    textDecoration: 'none',
    padding: '5px 0'
  },
  menuItem: {
    height: 'auto',
    padding: 0,
    color: '#333',
    cursor: 'pointer',
    width: '100%',
    textDecoration: 'none'
  },
  navIconHide: {
    [theme.breakpoints.up('md')]: {
      display: 'none'
    }
  },
  searchBox: {
    borderRadius: '5px',
    padding: '4px 0 2px 10px',
    border: '1px solid #ccc',
    textAlign: 'left'
  },
  toolbar: {
    backgroundColor: theme.palette.primary.main,
    color: '#fff',
    ...theme.mixins.toolbar
  }
});
