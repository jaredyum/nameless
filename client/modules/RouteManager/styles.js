export default {
  routeManager: {
    flex: 1,
    display: 'flex'
  }
};
