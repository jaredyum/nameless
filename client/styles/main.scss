// Vendors
@import './vendors/normalize';
@import './vendors/reset';

// Modules
@import './modules/defaults';
