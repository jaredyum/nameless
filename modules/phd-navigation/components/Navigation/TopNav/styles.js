export default theme => ({
  appBar: {
    zIndex: theme.zIndex.drawer + 1
  },
  navIconHide: {
    [theme.breakpoints.up('md')]: {
      display: 'none'
    }
  },
  flex: {
    flex: 1
  },
  searchBox: {
    borderRadius: '5px',
    padding: '4px 0 2px 10px',
    border: '1px solid #ccc',
    textAlign: 'left'
  }
});
