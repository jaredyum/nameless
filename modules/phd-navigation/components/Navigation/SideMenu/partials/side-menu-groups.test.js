import {
  getComponentWrapper,
  testCommonComponentAttrs
} from 'utils/tests';

// MUI Components
import MenuList from '@material-ui/core/MenuList';
import ListSubheader from '@material-ui/core/ListSubheader';

// Styles (the exact same styles your component uses)
import styles from '../../styles';

// Components
import SideMenuItems from './side-menu-items';
import SideMenuGroups from './side-menu-groups';

const defaultProps = {
  classes: styles,
  sideMenuConfig: [{
    navTitle: 'foo',
    navItems: [1, 2, 3]
  }, {
    navTitle: 'bar',
    navItems: [4, 5, 6]
  }]
};

let wrapper;

describe('<SideMenuGroups /> Component', () => {
  testCommonComponentAttrs(SideMenuGroups, defaultProps);

  beforeEach(() => {
    wrapper = getComponentWrapper(SideMenuGroups, defaultProps);
  });

  describe('basic rendering', () => {
    it('should render with as many MenuLists as are provided', () => {
      const menuLists = wrapper.find(MenuList);
      expect(menuLists).toHaveLength(2);
      // Assert that each list contains the correct child components.
      menuLists.forEach((menuList) => {
        expect(menuList.find(ListSubheader)).toHaveLength(1);
        expect(menuList.find(SideMenuItems)).toHaveLength(1);
      });
    });
  });
});
