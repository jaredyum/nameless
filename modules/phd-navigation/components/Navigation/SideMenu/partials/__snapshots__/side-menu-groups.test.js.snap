// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SideMenuGroups /> Component when rendering should render a component matching the snapshot 1`] = `
<div>
  <div
    key="foo"
  >
    <MenuList>
      <WithStyles(ListSubheader)>
        foo
      </WithStyles(ListSubheader)>
      <WithStyles(SideMenuItems)
        collection={
          Array [
            1,
            2,
            3,
          ]
        }
      />
    </MenuList>
    <WithStyles(Divider) />
  </div>
  <div
    key="bar"
  >
    <MenuList>
      <WithStyles(ListSubheader)>
        bar
      </WithStyles(ListSubheader)>
      <WithStyles(SideMenuItems)
        collection={
          Array [
            4,
            5,
            6,
          ]
        }
      />
    </MenuList>
  </div>
</div>
`;
